editor/example/spec/corces.ts

```diff
@@ -0,0 +1,275 @@
+                { type: 'filter', field: 'strand', oneOf: ['-'] }
+            ],
+            mark: 'text',
+            text: { field: 'name', type: 'nominal' },
+            x: { field: 'start', type: 'genomic' },
+            xe: { field: 'end', type: 'genomic' },
+            size: { value: 8 },
+            style: { textFontSize: 8, dy: 10 }
+        },
+        {
+            dataTransform: [
+                { type: 'filter', field: 'type', oneOf: ['gene'] },
+                { type: 'filter', field: 'strand', oneOf: ['+'] }
+            ],
+            mark: 'rect',
+            x: { field: 'end', type: 'genomic' },
+            size: { value: 7 }
+        },
+        {
+            dataTransform: [
+                { type: 'filter', field: 'type', oneOf: ['gene'] },
+                { type: 'filter', field: 'strand', oneOf: ['-'] }
+            ],
+            mark: 'rect',
+            x: { field: 'start', type: 'genomic' },
+            size: { value: 7 }
+        },
+        {
+            dataTransform: [{ type: 'filter', field: 'type', oneOf: ['exon'] }],
+            mark: 'rect',
+            x: { field: 'start', type: 'genomic' },
+            xe: { field: 'end', type: 'genomic' },
+            size: { value: 14 }
+        },
+        {
+            dataTransform: [{ type: 'filter', field: 'type', oneOf: ['gene'] }],
+            mark: 'rule',
+            x: { field: 'start', type: 'genomic' },
+            xe: { field: 'end', type: 'genomic' },
+            strokeWidth: { value: 3 }
+        }
+    ],
+    row: { field: 'strand', type: 'nominal', domain: ['+', '-'] },
+    color: { field: 'strand', type: 'nominal', domain: ['+', '-'], range: ['#012DB8', '#BE1E2C'] },
+    visibility: [
+        {
+            operation: 'less-than',
+            measure: 'width',
+            threshold: '|xe-x|',
+            transitionPadding: 10,
+            target: 'mark'
+        }
+    ],
+    width,
+    height: 80
+};
+
+const getBeddbData = (name: string) => {
+    return {
+        url: `https://server.gosling-lang.org/api/v1/tileset_info/?d=${name}-plac-seq-bedpe`,
+        type: 'beddb',
+        genomicFields: [
+            { name: 'start', index: 1 },
+            { name: 'end', index: 2 }
+        ]
+    };
+};
+
+const withinLinkTrack = [
+    { name: 'oligodendrocyte', color: '#F97E2A' },
+    { name: 'microglia', color: '#50ADF9' },
+    { name: 'neuron', color: '#7B0EDC' }
+].map((d, i) => {
+    return {
+        title: i == 0 ? 'PLAC-seq (H3K4me3) Nott et al.' : '',
+        data: getBeddbData(d.name),
+        mark: 'withinLink',
+        x: { field: 'start', type: 'genomic' },
+        xe: { field: 'end', type: 'genomic' },
+        y: { flip: true },
+        strokeWidth: { value: 1 },
+        color: { value: 'none' },
+        stroke: { value: d.color },
+        opacity: { value: 0.1 },
+        overlayOnPreviousTrack: i != 0,
+        width,
+        height: 60
+    };
+});
+
+const spec = {
```

Comment:
Could we explicitly define a type of individual variables used in JS-based examples? (e.g., GoslingSpec, DataDeep, OverlaidTrack)

Since our examples are becoming large, defining types will make it easier to maintain the examples (e.g., when we rename property names, IDE will automatically rename corresponding examples, etc).

```suggestion
const spec : GoslingSpec = {
```